integration_test/tests/free_and_cancel_tests.rs

```diff
@@ -0,0 +1,142 @@
+mod common;
+
+/// The tests in this module are based on the Preview Data workflow using SSIS.
+/// Although that tool is the inspiration for these tests, they more generally
+/// test what happens when
+///   1. a user allocates and uses a statement, and then calls SQLFreeStmt
+///   2. a user allocates and uses a statement to execute a query, and then
+///      calls SQLCancel
+/// These are workflows that could appear in any ODBC use-case, not just SSIS.
+mod integration {
+    use crate::common::{
+        allocate_env, connect_and_allocate_statement, disconnect_and_free_dbc_and_env_handles,
+        fetch_and_get_data, get_column_attributes,
+    };
+    use cstr::WideChar;
+    use definitions::{
+        CDataType, Desc, FreeStmtOption, HDbc, HEnv, HStmt, Handle, Pointer, SQLCancel,
+        SQLExecDirectW, SQLFreeStmt, SQLPrepareW, SQLRowCount, SQLSetStmtAttrW, SqlReturn,
+        StatementAttribute, SQL_NTS,
+    };
+
+    /// Setup flow that connects and allocates a statement. This allocates a
+    /// new environment handle, sets the ODBC_VERSION environment attribute,
+    /// connects using the default URI, and allocates a statement. The flow
+    /// is:
+    ///     - SQLAllocHandle(SQL_HANDLE_ENV)
+    ///     - SQLSetEnvAttr(SQL_ATTR_ODBC_VERSION, SQL_OV_ODBC3)
+    ///     - SQLAllocHandle(SQL_HANDLE_DBC)
+    ///     - SQLDriverConnectW
+    ///     - SQLAllocHandle(SQL_HANDLE_STMT)
+    fn setup_connect_and_alloc_stmt() -> (HEnv, HDbc, HStmt) {
+        let env_handle = allocate_env().unwrap();
+        let (conn_handle, stmt_handle) = connect_and_allocate_statement(env_handle, None);
+
+        (env_handle, conn_handle, stmt_handle)
+    }
+
+    /// This test is inspired by the SSIS Preview Data result set metadata flow.
+    /// It is altered to be more general than that specific flow, with a focus
+    /// on freeing the statement handle after gathering some metadata. This flow
+    /// depends on setup_connect_and_alloc_stmt.
+    /// After allocating a statement handle, the flow is:
+    ///     - SQLPrepareW(<query>)
+    ///     - SQLNumResultCols
+    ///     - <loop: for each column>
+    ///         - SQLDescribeColW
+    ///         - SQLColAttributeW(SQL_DESC_UNSIGNED)
+    ///         - SQLColAttributeW(SQL_DESC_UPDATABLE)
+    ///     - SQLFreeStmt(SQL_CLOSE)
+    ///     - SQLDisconnect
+    ///     - SQLFreeHandle(SQL_HANDLE_DBC)
+    ///     - SQLFreeHandle(SQL_HANDLE_ENV)
+    #[test]
+    fn test_free_stmt() {
```

Comment:
No need for more discussion. I see what you're saying. My initial idea here was that this was still a flow that made actual use of the statement handle (getting metadata) before freeing, but you're right that just calling `SQLPrepareW` is enough to demonstrate that point. I ended up leaving the name as `test_free_stmt` since I cut all the metadata-gathering stuff. Now it just does Prepare then frees the statement.